azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java

```diff
@@ -75,6 +77,22 @@
     public static final String UNKNOWN_DEPLOYMENT_TYPE = "The value of <deploymentType> is unknown, supported values are: " +
             "ftp, zip, msdeploy, run_from_blob and run_from_zip.";
 
+    public static final String APPINSIGHTS_INSTRUMENTATION_KEY = "APPINSIGHTS_INSTRUMENTATIONKEY";
+    public static final String FAILED_TO_GET_APPLICATION_INSIGHT = "Application insight %s was not found, " +
+            "will create it in resource group %s.";
+    public static final String APPLICATION_INSIGHTS_CREATE_FAILED = "Unable to create the Application Insights " +
+            "for the Function App. Please use the Azure Portal to manually create and configure the Application " +
+            "Insights if needed.";
+    public static final String APPLICATION_INSIGHTS_CREATE_START = "Creating application insights...";
+    public static final String APPLICATION_INSIGHTS_CREATED = "Successfully created the application insights %s " +
+            "for this Function App. You can visit https://portal.azure.com/#resource%s/overview to view your " +
+            "Application Insights component.";
+    public static final String SKIP_CREATING_APPLICATION_INSIGHTS = "Skip creating application insights";
+    public static final String APPLICATION_INSIGHTS_NOT_SUPPORTED = "Application Insights feature was not supported with" +
+            " current authentication, skip related steps.";
+    public static final String APP_INSIGHTS_CONFIGURATION_WAS_NOT_SUPPORTED = "Can not set appInsightsKey or " +
```

Comment:
Contradicting configuration for application insights,  specify 'appInsightsKey' or 'appInsightsInstance' if you want to enable it, and specify 'disableAppInsights=true' if you want to disable it.